pkg/utils/requestutils/curl/request.go

```diff
@@ -71,22 +77,23 @@ func (c *requestConfig) generateArgs() []string {
 	if c.verbose {
 		args = append(args, "-v")
 	}
-	if c.allowInsecure {
+	if c.ignoreServerCert {
 		args = append(args, "-k")
 	}
-	if c.withoutStats {
+	if c.silent {
 		args = append(args, "-s")
 	}
 	if c.connectionTimeout > 0 {
 		seconds := fmt.Sprintf("%v", c.connectionTimeout)
 		args = append(args, "--connect-timeout", seconds, "--max-time", seconds)
 	}
-	if c.returnHeaders {
+	if c.headersOnly {
 		args = append(args, "-I")
 	}
-	if c.method != http.MethodGet && c.method != "" {
-		args = append(args, "-X"+c.method)
-	}
+
+	// We prefer to be explicit, and always set the method
+	args = append(args, fmt.Sprintf("-X %s", c.method))
```

Comment:
mega-nit (not worth rerunning CI imo unless you're doing something else anyway):
```suggestion
	args = append(args, "-X", c.method)
```

Reply:
good call! updated